image_source_stream.go

```diff
@@ -2,10 +2,35 @@ package gostream
 
 import (
 	"context"
+	"errors"
+	"math"
+	"time"
 )
 
+type BackoffTuningOptions struct {
+	ExpBase          float64
+	Offset           float64
+	MaxSleepMilliSec float64
```

Comment:
MaxSleep and time.Duration

Reply:
i also suggested using duration initially, but it is a bit messy to work with since we need to convert back and forth for the exponential calculation bit.

Reply:
can we just convert internally once?